Bugzilla/Template.pm

```diff
@@ -227,10 +227,20 @@ sub renderComment {
     else {
         # We intentionally disable all html tags. Users should use markdown syntax.
         # This prevents things like inline styles on anchor tags, which otherwise would be valid.
+        # Note that this does not affect the html regex replacements that happens before this
+        # if block. See the comment about encoding hacks near the beginning of the function.
         $text =~ s/([<])/&lt;/g;
 
         # As a preference, we opt into all new line breaks being rendered as a new line.
         $text =~ s/(\r?\n)/  $1/g;
+
+        # Render text as markdown
+        $text = decode('UTF-8', Bugzilla->markdown_parser->render_html($text));
+
+        # At this point all links are plain links created by markdown, i.e. they have no additional
+        # html attributes and so our regex should match. Here we nofollow any non-bmo link.
+        my $urlbase = Bugzilla->localconfig->{urlbase};
+        $text =~ s/<a href="(?!$urlbase)/<a rel="nofollow" href="/g;
```

Comment:
I hope I'm not being naive about this - since we escape all html tags a few lines before this, I think it is safe to assume that any links at this point were created by the markdown parser above, which should output a plain link that matches this regex.

Reply:
This nofollows both autolinkifyied links and manual [ ]() markdown links.